### PR TITLE
add editorconfig and enhance/fix vagrant install/instructions

### PR DESCRIPTION
**Pull request by rubo77**
- add `.editorconfig` with indentation settings for the whole project
- update default node list URL
- Add some more troubleshooting info in README and bootstrap.sh
- enable automatic update check for the vagrant box
- add apt upgrade

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+indent_size = 4
+charset = utf-8
+
+[*.hbs]
+indent_style = space
+indent_size = 2
+
+[*.sh]
+indent_style = space
+indent_size = 2
```
Comment (RalfJung) [inline .editorconfig:17]: What is this file?

File: vagrant/Vagrantfile [hunk 2]
```diff
@@ -24,8 +24,8 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     t.vm.network :forwarded_port, guest: 8833, host: 8833
   end
   
-  # disable automatic update check:
-  config.vm.box_check_update = false
+  # enable automatic update check:
+  config.vm.box_check_update = true
```
Comment (RalfJung) [inline vagrant/Vagrantfile:28]: Why this change?

File: vagrant/bootstrap.sh [hunk 3]
```diff
@@ -34,7 +36,7 @@ sudo chown $FFNM_USERNAME .
 ffsudo="sudo -u $FFNM_USERNAME"
 
 : "#### We need some development libraries for the build process:"
-sudo apt update && sudo apt -y --no-install-recommends install git msmtp libc6-dev libssl-dev libpq-dev curl gcc pkg-config
+sudo apt update --allow-releaseinfo-change && sudo apt -y upgrade && sudo apt -y --no-install-recommends install git msmtp libc6-dev libssl-dev libpq-dev curl gcc pkg-config
```
Comment (RalfJung) [inline vagrant/bootstrap.sh:39]: "--allow-releaseinfo-change" seems rather suspicious.

File: .editorconfig (hunk 1 above)
Comment (rubo77) [reply .editorconfig:17]: it tells your editor how to style the indention of this project

File: vagrant/Vagrantfile (hunk 2 above)
Comment (rubo77) [reply vagrant/Vagrantfile:28]: I noticed that when debugging, why my vagrant setup didn't work, with this set= true, it just tells you, that you should update your vagrant buster image before (I had 9.99 and there is 10.0 out already). otherwise the apt-installation failed

File: vagrant/bootstrap.sh (hunk 3 above)
Comment (rubo77) [reply vagrant/bootstrap.sh:39]: I had buster 9.99 and there is 10.0 out already and without the option, `apt` cannot update. the option is harmless

File: .editorconfig (hunk 1 above)
Comment (RalfJung) [reply .editorconfig:17]: Hm, seems like a few editors are supported but many are not.

*shrug* I have no idea if this file here actually reflects my intentions, but I guess I'll complain when I see PRs with the wrong indentation. ;)

File: vagrant/bootstrap.sh (hunk 3 above)
Comment (RalfJung) [reply vagrant/bootstrap.sh:39]: The option AFAIK was added for security reasons. If you had 9.99 before that sounds like you were using a pre-release of buster or so. Outside dist upgrades, this option should not be needed, so I'd rather not add it here.
